modules/pipeline/providers/cron/cron.service.go

```diff
@@ -31,67 +33,78 @@ import (
+	if !apistructs.PipelineSource(req.PipelineSource).Valid() {
+		return nil, apierrors.ErrCreatePipelineCron.InvalidParameter(errors.Errorf("invalid pipelineSource: %s", req.PipelineSource))
 	}
-	if req.PipelineCreateRequest.PipelineYmlName == "" {
+	if req.PipelineYmlName == "" {
 		return nil, apierrors.ErrCreatePipelineCron.InvalidParameter(errors.Errorf("missing pipelineYmlName"))
 	}
-	if req.PipelineCreateRequest.PipelineYml == "" {
+	if req.PipelineYml == "" {
 		return nil, apierrors.ErrCreatePipelineCron.InvalidParameter(errors.Errorf("missing pipelineYml"))
 	}
-	pipelineYml, err := pipelineyml.New([]byte(req.PipelineCreateRequest.PipelineYml))
+
+	pipelineYml, err := pipelineyml.New([]byte(req.PipelineYml))
 	if err != nil {
-		return nil, err
+		return nil, apierrors.ErrParsePipelineYml.InternalError(err)
 	}
-	if pipelineYml.Spec().Cron == "" {
-		return nil, apierrors.ErrCreatePipelineCron.InvalidParameter(errors.Errorf("not cron pipeline"))
+	var compensator *apistructs.CronCompensator
+	if pipelineYml.Spec().CronCompensator != nil {
+		compensator = &apistructs.CronCompensator{}
+		compensator.Enable = pipelineYml.Spec().CronCompensator.Enable
+		compensator.StopIfLatterExecuted = pipelineYml.Spec().CronCompensator.StopIfLatterExecuted
+		compensator.LatestFirst = pipelineYml.Spec().CronCompensator.LatestFirst
 	}
 
-	// store to db
-	cron := db.PipelineCron{
-		PipelineSource:  apistructs.PipelineSource(req.PipelineCreateRequest.PipelineSource),
-		PipelineYmlName: req.PipelineCreateRequest.PipelineYmlName,
-		CronExpr:        pipelineYml.Spec().Cron,
-		Enable:          &[]bool{req.PipelineCreateRequest.AutoStartCron}[0],
-	}
-	if req.PipelineCreateRequest != nil {
-		var extra = db.PipelineCronExtra{
-			PipelineYml:  req.PipelineCreateRequest.PipelineYml,
-			ClusterName:  req.PipelineCreateRequest.ClusterName,
-			FilterLabels: req.PipelineCreateRequest.Labels,
-			NormalLabels: req.PipelineCreateRequest.NormalLabels,
-			Envs:         req.PipelineCreateRequest.Envs,
-			Version:      "v2",
-			Compensator: &apistructs.CronCompensator{
-				Enable: pipelineYml.Spec().CronCompensator.Enable,
-			},
-		}
-		cron.Extra.ConfigManageNamespaces = strutil.DedupSlice(append(cron.Extra.ConfigManageNamespaces, req.PipelineCreateRequest.ConfigManageNamespaces...), true)
-		cron.Extra.IncomingSecrets = req.PipelineCreateRequest.GetSecrets()
-
-		if pipelineYml.Spec().CronCompensator != nil {
-			cron.Extra.Compensator = &apistructs.CronCompensator{
-				Enable:               pipelineYml.Spec().CronCompensator.Enable,
-				LatestFirst:          pipelineYml.Spec().CronCompensator.LatestFirst,
-				StopIfLatterExecuted: pipelineYml.Spec().CronCompensator.StopIfLatterExecuted,
-			}
-		}
-
-		if req.PipelineCreateRequest.CronStartFrom != nil {
-			var cronStartFrom = req.PipelineCreateRequest.CronStartFrom.AsTime()
-			extra.CronStartFrom = &cronStartFrom
-		}
-		cron.Extra = extra
+	cron := &db.PipelineCron{
```

Comment:
bad var name, the same with pkg name. Your IDE has tip.
rename to `dbCron` or others.